### PR TITLE
replace vxrm with vcsr[vxrm]

### PR DESCRIPTION
**Pull request by rez5427**
When I run the vasub test file from the riscv-vector-test suite, I encountered a discrepancy between the outputs of Sail-RISCV and Spike, as shown in the image (highlighted in red). The issue stems from the function get_fixed_rounding_incr, which is incorrectly using the vxrm[1..0] bits instead of vcsr[vxrm]. The vxrm register is not being set by a CSR instruction; it is initialized to zero and remains zero throughout the execution, causing this mismatch.

image

### CODE REVIEW

**Comment by Timmmm** [comment]
Presumably this issue also exists for `vxsat`? I think we should also fix the duplicate read_csr and write_csr clauses at the same time.

**Comment by rez5427** [comment]
Is it appropriate to remove the registers vxrm and vxset in this context?

**Comment by Alasdair** [comment]
I don't think vxrm and vxsat should be actual Sail registers no, I think they are just parts of vcsr that should be accessible via the read_csr and write_csr functions.

File: model/riscv_sys_control.sail [hunk 1]
```diff
@@ -557,10 +557,6 @@ function init_sys() -> unit = {
    * See riscv_vlen.sail for details.
    */
   vstart             = zero_extend(0b0);
-  vxsat              = 0b0;
-  vxrm               = 0b00;
-  vcsr[vxrm]       = vxrm;
-  vcsr[vxsat]      = vxsat;
```
Comment (Alasdair) [inline model/riscv_sys_control.sail:563]: This might want to be `vcsr[vxrm] = 0b00` and `vcsr[vxsat] = 0b0` rather than removing both.

**Review by Alasdair — APPROVED** [review]
Note: we should make sure to squash these commits before merging.

File: model/riscv_insts_zicsr.sail [hunk 2]
```diff
@@ -71,8 +71,8 @@ function clause read_CSR(0b1011100 /* 0xB80 */ @ index : bits(5) if sizeof(xlen)
 
 /* vector */
 function clause read_CSR(0x008) = zero_extend(vstart)
-function clause read_CSR(0x009) = zero_extend(vxsat)
-function clause read_CSR(0x00A) = zero_extend(vxrm)
+function clause read_CSR(0x009) = zero_extend(vcsr[vxsat])
```
Comment (Timmmm) [inline model/riscv_insts_zicsr.sail:74]: I think we should probably fix #548 in this MR though. Move all the vector code from this file to `riscv_vext_control.sail` and then delete the duplicates (there are 3 copies for some of them!).

Note the existing code in that file seems to be correct - it includes this fix and also dirties the status flag which the code here doesn't do.

Comment (Timmmm) [reply model/riscv_insts_zicsr.sail:74]: Let me know if you can't be bothered though - I can have a go.

Comment (rez5427) [reply model/riscv_insts_zicsr.sail]: Sorry for the late reply. Is this okay?

Comment (Timmmm) [reply model/riscv_insts_zicsr.sail]: Yep LGTM, thanks!

**Review by Timmmm — APPROVED** [review]

**Pull request merged**